homeassistant/components/onvif/device.py

```diff
@@ -307,23 +322,31 @@ async def async_get_capabilities(self):
             self.device.create_imaging_service()
             imaging = True
 
-        events = False
+        return Capabilities(snapshot=snapshot, ptz=ptz, imaging=imaging)
+
+    async def async_start_events(self):
+        """Start the event handler."""
         with suppress(*GET_CAPABILITIES_EXCEPTIONS, XMLParseError):
             onvif_capabilities = self.onvif_capabilities or {}
             pull_point_support = onvif_capabilities.get("Events", {}).get(
                 "WSPullPointSupport"
             )
             LOGGER.debug("%s: WSPullPointSupport: %s", self.name, pull_point_support)
-            events = await self.events.async_start(
-                pull_point_support is not False, True
-            )
+            return await self.events.async_start(pull_point_support is not False, True)
 
-        return Capabilities(snapshot, events, ptz, imaging)
+        return False
 
     async def async_get_profiles(self) -> list[Profile]:
         """Obtain media profiles for this device."""
         media_service = self.device.create_media_service()
-        result = await media_service.GetProfiles()
+        LOGGER.debug("%s: xaddr for media_service: %s", self.name, media_service.xaddr)
```

Comment:
This is where the cheap-o camera was failing because it was returning the wrong xaddrs so I added logging for this.

I'll try to see if we can come up with a solution in the 2023.6.x cycle